samples/hello_events.c

```diff
@@ -89,59 +86,51 @@ struct ras_inject_error {
 
 // Global configuration of bus, set during parse_args()
 
-struct config{
+struct events_config {
 	struct target {
 		int bus;
 	} target;
 }
 
-config = {
+events_config = {
 	.target = {
 		.bus = -1
 	}
 };
 
-static fpga_result inject_ras_fatal_error(fpga_token fme_token, uint8_t err)
+fpga_result inject_ras_fatal_error(fpga_token fme_token, uint8_t err)
 {
-	fpga_result result                    = FPGA_OK;
-	fpga_handle fme_handle                = NULL;
-	struct ras_inject_error  inj_error    = { {0} };
-	fpga_object inj_err_object;
-
-	result = fpgaOpen(fme_token, &fme_handle, FPGA_OPEN_SHARED);
-	if (result != FPGA_OK) {
+	fpga_result             res1       = FPGA_OK;
+	fpga_result             res2       = FPGA_OK;
+	fpga_handle             fme_handle = NULL;
+	struct ras_inject_error inj_error  = { {0} };
+	fpga_object             inj_err_object;
+
+	res1 = fpgaOpen(fme_token, &fme_handle, FPGA_OPEN_SHARED);
+	if (res1 != FPGA_OK) {
 		OPAE_ERR("Failed to open FPGA");
-		return result;
+		return res1;
 	}
 
-	result = fpgaHandleGetObject(fme_handle, FME_SYSFS_INJECT_ERROR, &inj_err_object, 0);
-	if (result != FPGA_OK) {
-		OPAE_ERR("Failed to get Handle Object");
-		goto out_close;
-	}
+	res1 = fpgaHandleGetObject(fme_handle, FME_SYSFS_INJECT_ERROR, &inj_err_object, 0);
+	ON_ERR_GOTO(res1, out_close, "Failed to get Handle Object");
 
 	// Inject fatal error
 	inj_error.fatal_error = err;
 
-	result = fpgaObjectWrite64(inj_err_object, inj_error.csr, 0);
-	if (result != FPGA_OK) {
-		OPAE_ERR("Failed to Read Object ");
-		goto out_destroy_obj;
-	}
+	res1 = fpgaObjectWrite64(inj_err_object, inj_error.csr, 0);
+	ON_ERR_GOTO(res1, out_destroy_obj, "Failed to Read Object");
 
 out_destroy_obj:
-	result = fpgaDestroyObject(&inj_err_object);
-	if (result != FPGA_OK) {
-		OPAE_ERR("Failed to Destroy Object");
-		goto out_close;
-	}
-
+	res2 = fpgaDestroyObject(&inj_err_object);
+	ON_ERR_GOTO(res2, out_close, "Failed to Destroy Object");
```

Comment:
Shouldn't we just print a message here? We're `goto`'ing to a label that we would fall into next anyway, no?

Reply:
This does print an error, but there is no harm done is there?  I mean, it's logically correct.